conf/machine/include/imx-base.inc

```diff
@@ -23,11 +23,11 @@ IMX_DEFAULT_BSP_mx5 ?= "mainline"
 MACHINEOVERRIDES =. "use-${IMX_DEFAULT_BSP}-bsp:"
 
 PREFERRED_PROVIDER_u-boot ??= "${IMX_DEFAULT_BOOTLOADER}"
-PREFERRED_PROVIDER_u-boot-tools-native ??= "${IMX_DEFAULT_BOOTLOADER}-tools-native"
-PREFERRED_PROVIDER_nativesdk-u-boot-tools ??= "nativesdk-${IMX_DEFAULT_BOOTLOADER}-tools"
-PREFERRED_PROVIDER_u-boot-mkimage-native ??= "${IMX_DEFAULT_BOOTLOADER}-tools-native"
-PREFERRED_PROVIDER_nativesdk-u-boot-mkimage ??= "nativesdk-${IMX_DEFAULT_BOOTLOADER}-tools"
 PREFERRED_PROVIDER_virtual/bootloader ??= "${IMX_DEFAULT_BOOTLOADER}"
+PREFERRED_PROVIDER_u-boot-tools-native ??= "u-boot-imx-tools-native"
+PREFERRED_PROVIDER_nativesdk-u-boot-tools ??= "nativesdk-u-boot-imx-tools"
+PREFERRED_PROVIDER_u-boot-mkimage-native ??= "u-boot-imx-tools-native"
+PREFERRED_PROVIDER_nativesdk-u-boot-mkimage ??= "nativesdk-u-boot-imx-tools"
```

Comment:
We control the u-boot depending on the BSP we are using, so the change you are proposing does not improve it. Also, this might break customers' internal BSPs. Could you share the motivation for it?

Reply:
Just simply because IMX_DEFAULT_BOOTLOADER could be u-boot-fslc, in which case, ${IMX_DEFAULT_BOOTLOADER}-tools-native would make no sense because there is not a u-boot-fslc-tools-native recipe at all.

Reply:
This is right, but not necessarily the unique option as we have customers who do have it.

Reply:
Then maybe they should handle PREFERRED_PROVIDER in their own meta layers? Otherwise the customers dont have u-boot-fslc-tools recipes would be impacted? The truth is there is not a u-boot-fslc-tools recipe in meta-freescale or meta-freescale-3rdparty, then it should not be referred.

Reply:
By the way, without this patch, the end users will get this error:
```
NOTE: Multiple providers are available for u-boot-tools-native (u-boot-tools-native, u-boot-imx-tools-native)
Consider defining a PREFERRED_PROVIDER entry to match u-boot-tools-native
```
that's because PREFERRED_PROVIDER_u-boot-tools-native set to an invalid recipe so it does not take effect, means no PREFERRED_PROVIDER_u-boot-tools-native is being set.

Reply:
I mean for imx6/imx7 machines.